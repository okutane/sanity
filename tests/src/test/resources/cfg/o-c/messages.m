@interface MyObject
- (void)foo;
- (void)useSelf;
@end

@implementation MyObject
- (void)useSelf {
    [self foo];
}
@end

@interface MyChild : MyObject
@end

@implementation MyChild
- (void)foo {
    [super foo];
}
@end

MyObject *OBJ;

static void testNew() {
    OBJ = [MyObject new];
}

static void testAllocInit() {
    OBJ = [[MyObject alloc] init];
}