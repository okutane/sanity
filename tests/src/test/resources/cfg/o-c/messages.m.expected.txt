CFG: -[MyObject useSelf]
#0000:	 assign: *tmp0 = self
 - no source -
#0001:	 assign: *tmp1 = _cmd
 - no source -
#0002:	 assign: tmp2 = *tmp0
messages.m:8 [self foo];
#0003:	 call: tmp3 = objc_msgSend(tmp2, *OBJC_SELECTOR_REFERENCES_) <exit>
messages.m:8 [self foo];

CFG: -[MyChild foo]
#0000:	 assign: *tmp0 = self
 - no source -
#0001:	 assign: *tmp1 = _cmd
 - no source -
#0002:	 assign: tmp2 = *tmp0
messages.m:17 [super foo];
#0003:	 assign: *(tmp3.field0) = tmp2
messages.m:17 [super foo];
#0004:	 assign: tmp4 = *OBJC_CLASSLIST_SUP_REFS_$_
messages.m:17 [super foo];
#0005:	 assign: *(tmp3.field1) = tmp4
messages.m:17 [super foo];
#0006:	 call: tmp5 = objc_msgSendSuper2(tmp3, *OBJC_SELECTOR_REFERENCES_) <exit>
messages.m:17 [super foo];

CFG: testNew
#0000:	 assign: tmp0 = *OBJC_CLASSLIST_REFERENCES_$_
messages.m:24 OBJ = [MyObject new];
#0001:	 call: tmp1 = objc_msgSend(tmp0, *OBJC_SELECTOR_REFERENCES_.4)
messages.m:24 OBJ = [MyObject new];
#0002:	 assign: tmp2 = tmp1
messages.m:24 OBJ = [MyObject new];
#0003:	 assign: *OBJ = tmp2 <exit>
messages.m:24 OBJ = [MyObject new];

CFG: testAllocInit
#0000:	 assign: tmp0 = *OBJC_CLASSLIST_REFERENCES_$_
messages.m:28 OBJ = [[MyObject alloc] init];
#0001:	 call: tmp1 = objc_msgSend(tmp0, *OBJC_SELECTOR_REFERENCES_.6)
messages.m:28 OBJ = [[MyObject alloc] init];
#0002:	 call: tmp2 = objc_msgSend(tmp1, *OBJC_SELECTOR_REFERENCES_.8)
messages.m:28 OBJ = [[MyObject alloc] init];
#0003:	 assign: tmp3 = tmp2
messages.m:28 OBJ = [[MyObject alloc] init];
#0004:	 assign: *OBJ = tmp3 <exit>
messages.m:28 OBJ = [[MyObject alloc] init];

CFG: <module init>
#0000:	 assign: *OBJC_METH_VAR_NAME_ = "foo"
 - no source -
#0001:	 assign: *OBJC_SELECTOR_REFERENCES_ = OBJC_METH_VAR_NAME_
 - no source -
#0002:	 assign: *(*OBJC_CLASS_$_MyObject.field4) = l_OBJC_CLASS_RO_$_MyObject
 - no source -
#0003:	 assign: *(*OBJC_CLASS_$_MyObject.field3) = null
 - no source -
#0004:	 assign: *(*OBJC_CLASS_$_MyObject.field2) = _objc_empty_cache
 - no source -
#0005:	 assign: *(*OBJC_CLASS_$_MyObject.field1) = null
 - no source -
#0006:	 assign: *(*OBJC_CLASS_$_MyObject.field0) = OBJC_METACLASS_$_MyObject
 - no source -
#0007:	 assign: *(*OBJC_METACLASS_$_MyObject.field4) = l_OBJC_METACLASS_RO_$_MyObject
 - no source -
#0008:	 assign: *(*OBJC_METACLASS_$_MyObject.field3) = null
 - no source -
#0009:	 assign: *(*OBJC_METACLASS_$_MyObject.field2) = _objc_empty_cache
 - no source -
#0010:	 assign: *(*OBJC_METACLASS_$_MyObject.field1) = OBJC_CLASS_$_MyObject
 - no source -
#0011:	 assign: *(*OBJC_METACLASS_$_MyObject.field0) = OBJC_METACLASS_$_MyObject
 - no source -
#0012:	 assign: *OBJC_CLASS_NAME_ = "MyObject"
 - no source -
#0013:	 assign: *(*l_OBJC_METACLASS_RO_$_MyObject.field9) = null
 - no source -
#0014:	 assign: *(*l_OBJC_METACLASS_RO_$_MyObject.field8) = null
 - no source -
#0015:	 assign: *(*l_OBJC_METACLASS_RO_$_MyObject.field7) = null
 - no source -
#0016:	 assign: *(*l_OBJC_METACLASS_RO_$_MyObject.field6) = null
 - no source -
#0017:	 assign: *(*l_OBJC_METACLASS_RO_$_MyObject.field5) = null
 - no source -
#0018:	 assign: *(*l_OBJC_METACLASS_RO_$_MyObject.field4) = OBJC_CLASS_NAME_
 - no source -
#0019:	 assign: *(*l_OBJC_METACLASS_RO_$_MyObject.field3) = null
 - no source -
#0020:	 assign: *(*l_OBJC_METACLASS_RO_$_MyObject.field2) = 40
 - no source -
#0021:	 assign: *(*l_OBJC_METACLASS_RO_$_MyObject.field1) = 40
 - no source -
#0022:	 assign: *(*l_OBJC_METACLASS_RO_$_MyObject.field0) = 3
 - no source -
#0023:	 assign: *OBJC_METH_VAR_NAME_.1 = "useSelf"
 - no source -
#0024:	 assign: *OBJC_METH_VAR_TYPE_ = "v16@0:8"
 - no source -
#0025:	 assign: *(*l_OBJC_CLASS_RO_$_MyObject.field9) = null
 - no source -
#0026:	 assign: *(*l_OBJC_CLASS_RO_$_MyObject.field8) = null
 - no source -
#0027:	 assign: *(*l_OBJC_CLASS_RO_$_MyObject.field7) = null
 - no source -
#0028:	 assign: *(*l_OBJC_CLASS_RO_$_MyObject.field6) = null
 - no source -
#0029:	 assign: *(*l_OBJC_CLASS_RO_$_MyObject.field5) = l_OBJC_$_INSTANCE_METHODS_MyObject
 - no source -
#0030:	 assign: *(*l_OBJC_CLASS_RO_$_MyObject.field4) = OBJC_CLASS_NAME_
 - no source -
#0031:	 assign: *(*l_OBJC_CLASS_RO_$_MyObject.field3) = null
 - no source -
#0032:	 assign: *(*l_OBJC_CLASS_RO_$_MyObject.field2) = 0
 - no source -
#0033:	 assign: *(*l_OBJC_CLASS_RO_$_MyObject.field1) = 0
 - no source -
#0034:	 assign: *(*l_OBJC_CLASS_RO_$_MyObject.field0) = 2
 - no source -
#0035:	 assign: *(*OBJC_CLASS_$_MyChild.field4) = l_OBJC_CLASS_RO_$_MyChild
 - no source -
#0036:	 assign: *(*OBJC_CLASS_$_MyChild.field3) = null
 - no source -
#0037:	 assign: *(*OBJC_CLASS_$_MyChild.field2) = _objc_empty_cache
 - no source -
#0038:	 assign: *(*OBJC_CLASS_$_MyChild.field1) = OBJC_CLASS_$_MyObject
 - no source -
#0039:	 assign: *(*OBJC_CLASS_$_MyChild.field0) = OBJC_METACLASS_$_MyChild
 - no source -
#0040:	 assign: *OBJC_CLASSLIST_SUP_REFS_$_ = OBJC_CLASS_$_MyChild
 - no source -
#0041:	 assign: *OBJC_CLASS_NAME_.2 = "MyChild"
 - no source -
#0042:	 assign: *(*l_OBJC_METACLASS_RO_$_MyChild.field9) = null
 - no source -
#0043:	 assign: *(*l_OBJC_METACLASS_RO_$_MyChild.field8) = null
 - no source -
#0044:	 assign: *(*l_OBJC_METACLASS_RO_$_MyChild.field7) = null
 - no source -
#0045:	 assign: *(*l_OBJC_METACLASS_RO_$_MyChild.field6) = null
 - no source -
#0046:	 assign: *(*l_OBJC_METACLASS_RO_$_MyChild.field5) = null
 - no source -
#0047:	 assign: *(*l_OBJC_METACLASS_RO_$_MyChild.field4) = OBJC_CLASS_NAME_.2
 - no source -
#0048:	 assign: *(*l_OBJC_METACLASS_RO_$_MyChild.field3) = null
 - no source -
#0049:	 assign: *(*l_OBJC_METACLASS_RO_$_MyChild.field2) = 40
 - no source -
#0050:	 assign: *(*l_OBJC_METACLASS_RO_$_MyChild.field1) = 40
 - no source -
#0051:	 assign: *(*l_OBJC_METACLASS_RO_$_MyChild.field0) = 1
 - no source -
#0052:	 assign: *(*OBJC_METACLASS_$_MyChild.field4) = l_OBJC_METACLASS_RO_$_MyChild
 - no source -
#0053:	 assign: *(*OBJC_METACLASS_$_MyChild.field3) = null
 - no source -
#0054:	 assign: *(*OBJC_METACLASS_$_MyChild.field2) = _objc_empty_cache
 - no source -
#0055:	 assign: *(*OBJC_METACLASS_$_MyChild.field1) = OBJC_METACLASS_$_MyObject
 - no source -
#0056:	 assign: *(*OBJC_METACLASS_$_MyChild.field0) = OBJC_METACLASS_$_MyObject
 - no source -
#0057:	 assign: *(*l_OBJC_CLASS_RO_$_MyChild.field9) = null
 - no source -
#0058:	 assign: *(*l_OBJC_CLASS_RO_$_MyChild.field8) = null
 - no source -
#0059:	 assign: *(*l_OBJC_CLASS_RO_$_MyChild.field7) = null
 - no source -
#0060:	 assign: *(*l_OBJC_CLASS_RO_$_MyChild.field6) = null
 - no source -
#0061:	 assign: *(*l_OBJC_CLASS_RO_$_MyChild.field5) = l_OBJC_$_INSTANCE_METHODS_MyChild
 - no source -
#0062:	 assign: *(*l_OBJC_CLASS_RO_$_MyChild.field4) = OBJC_CLASS_NAME_.2
 - no source -
#0063:	 assign: *(*l_OBJC_CLASS_RO_$_MyChild.field3) = null
 - no source -
#0064:	 assign: *(*l_OBJC_CLASS_RO_$_MyChild.field2) = 0
 - no source -
#0065:	 assign: *(*l_OBJC_CLASS_RO_$_MyChild.field1) = 0
 - no source -
#0066:	 assign: *(*l_OBJC_CLASS_RO_$_MyChild.field0) = 0
 - no source -
#0067:	 assign: *OBJC_CLASSLIST_REFERENCES_$_ = OBJC_CLASS_$_MyObject
 - no source -
#0068:	 assign: *OBJC_METH_VAR_NAME_.3 = "new"
 - no source -
#0069:	 assign: *OBJC_SELECTOR_REFERENCES_.4 = OBJC_METH_VAR_NAME_.3
 - no source -
#0070:	 assign: *OBJ = null
 - no source -
#0071:	 assign: *OBJC_METH_VAR_NAME_.5 = "alloc"
 - no source -
#0072:	 assign: *OBJC_SELECTOR_REFERENCES_.6 = OBJC_METH_VAR_NAME_.5
 - no source -
#0073:	 assign: *OBJC_METH_VAR_NAME_.7 = "init"
 - no source -
#0074:	 assign: *OBJC_SELECTOR_REFERENCES_.8 = OBJC_METH_VAR_NAME_.7
 - no source -
#0075:	 assign: *(*OBJC_LABEL_CLASS_$+1) = OBJC_CLASS_$_MyChild
 - no source -
#0076:	 assign: **OBJC_LABEL_CLASS_$ = OBJC_CLASS_$_MyObject
 - no source -
#0077:	 assign: *(*llvm.compiler.used+16) = OBJC_LABEL_CLASS_$
 - no source -
#0078:	 assign: *(*llvm.compiler.used+15) = OBJC_SELECTOR_REFERENCES_.8
 - no source -
#0079:	 assign: *(*llvm.compiler.used+14) = OBJC_METH_VAR_NAME_.7
 - no source -
#0080:	 assign: *(*llvm.compiler.used+13) = OBJC_SELECTOR_REFERENCES_.6
 - no source -
#0081:	 assign: *(*llvm.compiler.used+12) = OBJC_METH_VAR_NAME_.5
 - no source -
#0082:	 assign: *(*llvm.compiler.used+11) = OBJC_SELECTOR_REFERENCES_.4
 - no source -
#0083:	 assign: *(*llvm.compiler.used+10) = OBJC_METH_VAR_NAME_.3
 - no source -
#0084:	 assign: *(*llvm.compiler.used+9) = OBJC_CLASSLIST_REFERENCES_$_
 - no source -
#0085:	 assign: *(*llvm.compiler.used+8) = l_OBJC_$_INSTANCE_METHODS_MyChild
 - no source -
#0086:	 assign: *(*llvm.compiler.used+7) = OBJC_CLASS_NAME_.2
 - no source -
#0087:	 assign: *(*llvm.compiler.used+6) = OBJC_CLASSLIST_SUP_REFS_$_
 - no source -
#0088:	 assign: *(*llvm.compiler.used+5) = l_OBJC_$_INSTANCE_METHODS_MyObject
 - no source -
#0089:	 assign: *(*llvm.compiler.used+4) = OBJC_METH_VAR_TYPE_
 - no source -
#0090:	 assign: *(*llvm.compiler.used+3) = OBJC_METH_VAR_NAME_.1
 - no source -
#0091:	 assign: *(*llvm.compiler.used+2) = OBJC_CLASS_NAME_
 - no source -
#0092:	 assign: *(*llvm.compiler.used+1) = OBJC_SELECTOR_REFERENCES_
 - no source -
#0093:	 assign: **llvm.compiler.used = OBJC_METH_VAR_NAME_ <exit>
 - no source -

